views/default/js/mentions/autocomplete.js

```diff
@@ -87,23 +95,42 @@ define(function(require) {
 		});
 	};
 
-	var autocomplete = function (content, position) {
-		beforeMention = content.substring(0, position);
-		afterMention = content.substring(position);
-		parts = beforeMention.split(' ');
-		current = parts[parts.length - 1];
-
-		precurrent = false;
-		if (parts.length > 1) {
-			precurrent = parts[parts.length - 2];
-			if (!current.match(/@/)) {
-				if (precurrent.match(/@/)) {
-					current = precurrent + ' ' + current;
+	var autocomplete = function (content, position_or_data) {
+		if (mentionsEditor == 'ckeditor') {
+			edata = position_or_data;
+			current = "";
+			if (content.length != 0) {
+				//Split to get last item
+				var words = content.split(/ +/);
+				var html_words = edata.split(/ +|\r\n|\n/);
+				var last_word = words[words.length -1];
+				var last_html_word = html_words[html_words.length -2];
+				if (last_word.match(/^@\w{3,}/)&& (last_html_word == last_word+'</p>' || last_html_word == '<p>'+last_word+'</p>')) {
```

Comment:
What about other html tags? Perhaps we just need a nice regex to grab the last word that is wrapper or unwrapped.